ui/design-system-core/src/components/Icon/index.tsx

```diff
@@ -18,6 +18,7 @@ export interface IconProps {
   disabled?: boolean;
   testId?: string;
   styling?: string;
+  onClick?: (event: React.SyntheticEvent<Element, Event>) => void;
```

Comment:
```suggestion
```

Let's keep the API of the icon component lean and clean. You can handle the `onClick` event as follows:

```
<button onClick={...}>
    <Icon .../>
</button>
```

Once I finish the work on enhancing the button component I will replace the above suggestion with something like :

```
<Button onClick={...}>
    <Icon .../>
</Button>
```

It's a better API design to wrap every design system component which needs the `onClick` handler with the button element or component. That way we maintain the semantic meaning of each component. Buttons handle click events and the other components address a specific concern. In the case of icons the concern is simply to render icons of different sizes and styles.

Reply:
Thanks, Didd. I have made the change.

Reply:
Thanks, Didd. I have made the change.

Reply:
But why is it a better design like this? Is there any specific advantage of keeping semantic purity that outweighs the hassle of having to wrap components like divs or icons  whenever you want a simple click handler? @didd

Reply:
@quininez what makes it better API design is to gain the benefits of better SEO and accessibility as a result of using semantic HTML. As a result of using button we can gain a native support for keyboard navigation, better support for screen readers, we can get a native support for disabled state of an element, better unit testing experience and the like. 

And in this PR's case we have an icon button which we can use but the button component will need to be enhanced first then we can have a choice of using the onClick handler as 
`<Button icon={...} onClick={...} ...` or 
`<Button onClick={...}> <Icon .../> </Button> `

Reply:
That's a nice explanation Didd 👍